DKGRituals/src/coordinator.ts

```diff
@@ -106,7 +107,11 @@ export function handleEndRitual(event: EndRitualEvent): void {
   }
 
   ritualCounter.notEnded = ritualCounter.notEnded - 1;
-  ritualCounter.ended = ritualCounter.ended + 1;
+  if (event.params.successful) {
+    ritualCounter.successful = ritualCounter.successful + 1;
+  } else {
+    ritualCounter.unsuccessful = ritualCounter.unsuccessful + 1;
```

Comment:
@vzotova / @cygnusv is it possible that we get multiple EndRitual events emitted when unsuccessful - basically if there is an aggregation mismatch for multiple nodes submitting their aggregated transcript, can multiple EndRitual events get emitted? This code - https://github.com/nucypher/nucypher-contracts/blob/main/contracts/contracts/coordination/Coordinator.sol#L394.

If so, then this calc ends up counting more than it is supposed to.

Reply:
Good question! I think it should not be possible since a precondition to post aggregations is that the ritual is in the WAITING_FOR_AGGREGATIONS phase, and once there's a mismatch, there status should change to `DKG_INVALID`. See https://github.com/nucypher/nucypher-contracts/blob/c923625e187422e9033aa5690306b7b022cf96dc/contracts/contracts/coordination/Coordinator.sol#L359

Reply:
Ah right. Perfect!

Appreciate the clarification. @manumonti you can ignore my concern.